datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/ZookeeperBackedDatastreamStore.java

```diff
@@ -127,6 +128,12 @@ public void updateDatastream(String key, Datastream datastream, boolean notifyLe
       throw new DatastreamException("Datastream does not exists, can not be updated: " + key);
     }
 
+    // As this limit is ZK specific, adding this validation check specifically in ZookeeperBackedDatastreamStore.
+    double datastreamBlobSizeInMBs = getBlobSizeInMBs(DatastreamUtils.toJSON(datastream));
+    Validate.isTrue(datastreamBlobSizeInMBs <= ZNODE_BLOB_SIZE_LIMIT,
```

Comment:
The validation check is tricky, especially in the case of a Programmatic update. This will block the datastream update until the logic in the caller is fixed. Can you think of a less disruptive way?

Reply:
What are your concerns @vmaheshw? I understand that we need to add similar validation on the client side as well. If a client request is breaching the ZK node size limit, it will eventually fail the update anyway, it's just the failure will happen in the ZkAdapter/ZkClient layer. This to me is a more explicit and descriptive way to fail.

Reply:
I initially thought that it can impact the datastream restart, but I was wrong.

However, in the scenario, the limit goes >1MB because of the violation list, we will not be able to disable this at the server level and will have to rely on the external service to disable this feature. Until then, any other update/allowlisting will not go through.

Reply:
@vmaheshw understood your concern. 

I have added steps in both update and create paths to only adhere to the throughputViolatingTopic metadata if the corresponding config is enabled in our server.